### PR TITLE
Add valgrind tests to Travis

### PR DESCRIPTION
**Pull request by hjsuh18**
Added Valgrind tests cloned from bitbucket repository to Travis. 4 cron jobs are set each running different tests to avoid the 50 minute time limit on Travis.

### CODE REVIEW

**Comment by RobAtticus** [comment]
update: running all regression tests at once takes too long, so we'll explore splitting into 2 jobs.

File: .travis.yml [hunk 1]
```diff
@@ -46,6 +46,19 @@ jobs:
         # Versions <=9.6.1 is missing function `convert_tuples_by_name_map` needed by our code
         - PG_MAJOR=9.6 PG_MINOR_COMPILE=2 bash -x ./scripts/docker-run-abi-test.sh
 
+    - if: type = cron
+      stage: test
+      install: travis_wait 30 mvn install
+      env:
+        - PG_VERSION=10.0 PG_GIT_TAG=REL_10_0
+        - secure: "uGX2k6j4U6rIIw/+/8Nj2CfVMN6nc0mMhvFK2w1ADRm93Cr3rYo29vfCJYpDGn4W/UIpdl3r15KDJuC/8xF6waHTc9qFpXmUi/VeDLsnsEvBjf3cbK/rKfuzAYZsrRldbwCm+jHUN/aV8q43eGTcrffCFfuhe2gr+sOACCqs/ZxrZlvH2vfPj7pH+FY2xY/J6NvMvmZJljc8rbDJHvuBOu/IIEnhzXNRgOehc7INcaSZM34rx6ktlArJhFWuKnWLfpwXg6v/7So01kGSC0tB9gmL23eXCDrNbIqFrsFn130EfiVUQVClA+j6PGc1N7w+0QWm4h7TlGwwNgthBRnHxlskq4Fkal9hAwqup9Lp43567iaPM4SBJJbksw8y10tucgxWYpRcJU41FAGSc6BJwty82nzbOQLA3i8M95Xyp7ppa5KK4t2ay6818v3zZtID6oX8jVvodDQVHF3QbbhIuKXi+mhTK1RlsCOZ/SBc7ABtPKYxBcsclZHW17qkGbhXZyaHjwtWtzScHNXBg4jLzT/W89vvS654/sp9OpmXkO0g1XGAPAk525JQPrre08gQvkiSi2D56JDsu8XvRkZd7utrzEbpx+3oBW5W07eGykcWd6vD6n15tJZD5o2AUcHr8SjmiVdxvg1iajNzDuoznMUMl/5pgl+m+IHbdNxO4lQ="
```
Comment (cevian) [inline .travis.yml]: What are these two different secure keys. Should we add a comment to what each of these are for?

**Comment by RobAtticus** [comment]
@cevian This PR is updated. We ended up breaking it up into 4 jobs since two tests each take 30+ mins by themselves. So each one has its own job, and then we split the rest into two parts. 

All of these test on PG9.6 currently, we can look into PG10 testing in the future.

**Comment by RobAtticus** [comment]
Here is the build where they all pass for reference (this PR won't run them again since they are cron jobs):
https://travis-ci.org/timescale/timescaledb/builds/409106996

File: .travis.yml [hunk 2]
```diff
@@ -46,6 +46,74 @@ jobs:
...
+      before_script:
+        # clone valgrind test script
+        - git clone https://$USR:$PASSWORD@bitbucket.org/440-labs/tsdb-dev-tools.git /tmp/tsdb-dev-tools
+        - bash -c 'while true; do echo -e "keepalive\n"; sleep 60; done' & # print line every 60 minutes to avoid Travis timeout
+      script:
+        # TEST_MAX specifies the maximum test # to go up to
+        - cd /tmp/tsdb-dev-tools; TIMESCALE_DIR=$TRAVIS_BUILD_DIR TEST_MAX=40 EXCLUDE_PATTERN='plan_hashagg_optimized*|plan_hashagg_results*' bash -x ./test_valgrind.sh
```
Comment (erimatnor) [inline .travis.yml]: This command and many others run bash with `-x` which outputs every command in the log, which risks exposing secret variables. This is a highly discouraged practice with secure variables, including using `set -x` in scripts. See here: https://docs.travis-ci.com/user/best-practices-security/

Should probably make sure that no scripts are run with `-x` and also check that scripts that are run doesn't have `set -x` in them (including in the checked out dev-tools repo).

**Comment by hjsuh18** [comment]
@RobAtticus @cevian @erimatnor 
This PR has been updated taking comments into account.

File: .travis.yml [hunk 3]
```diff
@@ -52,6 +52,74 @@ jobs:
         # Version >= 9.6.3 is required by TimescaleDB
         - PG_MAJOR=9.6 PG_MINOR_COMPILE=3 bash -x ./scripts/docker-run-abi-test.sh
 
+    # Valgrind test #1 : first 40 excluding plan_hashagg_optimized* and plan_hashagg_results*
+    - if: type = cron
+      stage: test
+      env:
+        - PG_VERSION=9.6.6 PG_GIT_TAG=REL9_6_6
+        - secure: "TurskgAodG44RKCUPfoUb4JlT+UZwaPjUN73MS4+2ED2ayf46iUpcMODovDf7CPQRriMAvii/5xQ9UWTGYZejJy/Si3aE0fQeSKHIOPmuHqmmBkC3jgK9A5vOMpaAwljSNvIAMfwhIjVsq87nyvxHYWReDWl5SrdxTvh91RJvOiMRNdqwBbnj2j1cC30783ICGZUazrUyhDDnXispyBJYFxlHJmQJpfs6428dTRNJY/X8nVDXFwQkRjlIXarquTsrifCfzdq40Ykb8/1CS0295R8/kdKamQqXM0aS7iIkw3JFmZIz08OHKc7wcJuv6KSnCfNSftJqy2+nb1gkyIqkr3mrpYUa+2tKVwpA6OfET2S2kC5B3bOTcyBgihhnivAOS9zG8PTouLSUzqZzo+H893xdaMlX4kX3oOrOqzbj8YvTw6Dgw8PWERfnOngpKmKJFb8yboXPmGY7luSVxOplnPnYO0jaAXPU4ESsFYL3EBG5Zo4rVIuA5g7aJ0Ex+e5GQWiHDM1ubUo2vcuVVlurs/Aj8eSpNhqo0S5MjI05Jcz82X+9+asFcP+QV81UrueiXTlQ/7buAyMyVtKOgOhZwR4ImA1zzWkJjNNbZMJqeO9tPEgU7IxrqLhMgIpBB/pn0gzJZ8LN7C2oyznMYGzS5lrYaavojFUViv8vFyAQ24="
```
Comment (cevian) [inline .travis.yml]: Can these secure be shared across jobs, so don't have to be repeated?

Comment (RobAtticus) [reply .travis.yml]: They can @hjsuh18 . You want replace the top `env` section to look like this:
```
env:
  global:
    - secure: "TurskgAodG44RKCUPfoUb4JlT+UZwaPjUN73MS4+2ED2ayf46iUpcMODovDf7CPQRriMAvii/5xQ9UWTGYZejJy/Si3aE0fQeSKHIOPmuHqmmBkC3jgK9A5vOMpaAwljSNvIAMfwhIjVsq87nyvxHYWReDWl5SrdxTvh91RJvOiMRNdqwBbnj2j1cC30783ICGZUazrUyhDDnXispyBJYFxlHJmQJpfs6428dTRNJY/X8nVDXFwQkRjlIXarquTsrifCfzdq40Ykb8/1CS0295R8/kdKamQqXM0aS7iIkw3JFmZIz08OHKc7wcJuv6KSnCfNSftJqy2+nb1gkyIqkr3mrpYUa+2tKVwpA6OfET2S2kC5B3bOTcyBgihhnivAOS9zG8PTouLSUzqZzo+H893xdaMlX4kX3oOrOqzbj8YvTw6Dgw8PWERfnOngpKmKJFb8yboXPmGY7luSVxOplnPnYO0jaAXPU4ESsFYL3EBG5Zo4rVIuA5g7aJ0Ex+e5GQWiHDM1ubUo2vcuVVlurs/Aj8eSpNhqo0S5MjI05Jcz82X+9+asFcP+QV81UrueiXTlQ/7buAyMyVtKOgOhZwR4ImA1zzWkJjNNbZMJqeO9tPEgU7IxrqLhMgIpBB/pn0gzJZ8LN7C2oyznMYGzS5lrYaavojFUViv8vFyAQ24="
    - secure: "S8WfPsrVGg21vZPxjrmy1X19T7zpHNgaSAoqYGzJdDZcNYuZbI87+tP3VQZo4MxjP3Ag46zTI5K+6aMjCqYqFGKufZdEaSpPSLmppFbXALgSp/rOkzp1M9HhGj0QVP7QcoPbva5Bjnj3zVi9CMAo6MwtnGAvzqBLOsdtnXTxEmKYyeNgwUVbQi2nzO38GeLgabaFL4XD3f3bmHE5jG6QAwZ/L7+jajsD1biH4sOiRLCRshtvermtYxh3SK6rML2kiwATABxTd1xDdMVRO/llmDBN373tHK6mufJyIhrwG9oTprSSDDIJlEbSPCkH0uvJBGcDvOAWf9tr4TGs/beawx3ELcXqR/4EisFqvMiVL4Vpt1gMBg9gue4Y0wpHZYoBOOef2gFtuyL7PjYe9koYJAJxg9DJ29DMxIjuCLTnbHN6yLa9425pdvxcuNEJ2K2zD0ZWoLSVueL0MxLhOZJsqlMkmmrFyI7y5cj1XvJF3vcsa5yK3S7sXroKUzeLu6QMh8hO3jD/2IvdQJ7Huy5uiJk2a+KUceLzruxLbwPF4b8SEoTquJ9NFGtagCG77lbIbLltLs2fUf+JnM8ipdpoMibVRFBOahP8NZwpL/NFzG3WLH3tDncj/c3fneMiNv3FUHLf6ufOVOayQiAEidszcMct2R20qIxggXKymrjZA3k="
  matrix:
    # PG_VERSION (docker) and PG_GIT_TAG (Git) should be the same
    - PG_VERSION=9.6.6 PG_GIT_TAG=REL9_6_6
    - PG_VERSION=10.2 PG_GIT_TAG=REL_10_2
```

**Comment by hjsuh18** [comment]
I've made the changes, but I'm worrying that because we have not yet found a way to integrate valgrind testing with Travis for PG_VERSION 10, this will be problematic. From my understanding, the matrix env variable runs the job twice, once with version 9 and once with version 10. If this is the case, the tests will fail with version 10. 
Will test out to make sure.

**Comment by RobAtticus** [comment]
@hjsuh18 I tested on the `valgrind_test` branch with the updated `env` vars and it did not seem to run Valgrind tests twice.

**Comment by hjsuh18** [comment]
Yep, not sure 100% how it decides to run with version 9, but it seems to work fine. Should be good to go.

**Pull request merged**